src/main/java/org/usfirst/frc4904/robot/commands/controlpanel/ColorSensor.java

```diff
@@ -0,0 +1,223 @@
+`1¶¶¶¶¶¶¶1_1_01__`01```_1```_1__1_11___1_``00¶1````````
+``¶¶¶¶¶¶¶0`__10__000````1____1____1___1_```10¶0_```````
+``0¶¶¶¶¶¶¶1___0000000```11___1__`_0111_```000¶01```````
+```¶¶¶00000¶¶¶¶¶¶¶¶¶01___1___00_1¶¶¶`_``1¶¶10¶¶0```````
+```1010000¶000¶¶0100_11__1011000¶¶0¶1_10¶¶¶_0¶¶00``````
+10¶000000000¶0________0¶000000¶¶0000¶¶¶¶000_0¶0¶00`````
+¶¶¶¶¶¶0000¶¶¶¶_`___`_0¶¶¶¶¶¶¶00000000000000_0¶00¶01````
+¶¶¶¶¶0¶¶¶¶¶¶¶¶¶_``_1¶¶¶00000000000000000000_0¶000¶01```
+1__```1¶¶¶¶¶¶¶¶¶00¶¶¶¶00000000000000000000¶_0¶0000¶0_``
+```````¶¶¶¶¶¶¶¶¶¶¶¶¶¶¶00000000000000000000010¶00000¶¶_`
+```````0¶¶¶¶¶¶¶¶¶¶¶¶¶¶00000000000000000000¶10¶¶0¶¶¶¶¶0`
+````````¶¶¶¶¶¶¶¶¶¶0¶¶¶00000000000000000000010¶¶¶0011```
+````````1¶¶¶¶¶¶¶¶¶¶0¶¶¶0000000000000000000¶100__1_`````
+`````````¶¶¶¶¶¶¶¶¶¶¶¶¶¶¶000000000000000000¶11``_1``````
+`````````1¶¶¶¶¶¶¶¶¶¶¶0¶¶¶00000000000000000¶11___1_`````
+``````````¶¶¶¶¶¶0¶0¶¶¶¶¶¶¶0000000000000000¶11__``1_````
+``````````¶¶¶¶¶¶¶0¶¶¶0¶¶¶¶¶000000000000000¶1__````__```
+``````````¶¶¶¶¶¶¶¶0¶¶¶¶¶¶¶¶¶0000000000000000__`````11``
+`````````_¶¶¶¶¶¶¶¶¶000¶¶¶¶¶¶¶¶000000000000011_``_1¶¶¶0`
+`````````_¶¶¶¶¶¶0¶¶000000¶¶¶¶¶¶¶000000000000100¶¶¶¶0_`_
+`````````1¶¶¶¶¶0¶¶¶000000000¶¶¶¶¶¶000000000¶00¶¶01`````
+`````````¶¶¶¶¶0¶0¶¶¶0000000000000¶0¶00000000011_``````_
+````````1¶¶0¶¶¶0¶¶¶¶¶¶¶000000000000000000000¶11___11111
+````````¶¶¶¶0¶¶¶¶¶00¶¶¶¶¶¶000000000000000000¶011111111_
+```````_¶¶¶¶¶¶¶¶¶0000000¶0¶00000000000000000¶01_1111111
+```````0¶¶¶¶¶¶¶¶¶000000000000000000000000000¶01___`````
+```````¶¶¶¶¶¶0¶¶¶000000000000000000000000000¶01___1````
+``````_¶¶¶¶¶¶¶¶¶00000000000000000000000000000011_111```
+``````0¶¶0¶¶¶0¶¶0000000000000000000000000000¶01`1_11_``
+``````¶¶¶¶¶¶0¶¶¶0000000000000000000000000000001`_0_11_`
+``````¶¶¶¶¶¶¶¶¶00000000000000000000000000000¶01``_0_11`
+``````¶¶¶¶0¶¶¶¶00000000000000000000000000000001```_1_11
+ */
+public abstract class ColorSensor extends SequentialCommandGroup {
+    public final Color[] colorOrder = new Color[] { Color.RED, Color.YELLOW, Color.BLUE, Color.GREEN };
+
+    public Color currentColor = null;
+    private int colorsPassed = 0;
+    public boolean doneSpinning = false;
+
+    public void trackColorChange() {
+        if (currentColor == null) {
+            currentColor = getColor();
+            return;
+        }
+        Color tempColor = getColor();
+        if (currentColor != tempColor) {
+            Color oldColor = currentColor;
+            currentColor = tempColor;
+            colorsPassed++;
+            if (getIndex(currentColor) == 3) {
+                if (currentColor == colorOrder[0]) {
+                    if (colorsPassed == 23) {
+                        doneSpinning = true;
+                    }
+                } else {
+                    LogKitten.wtf("A color somehow was skipped and we have no code to handle it! :)");
+                    // it is very unlikely for a color to be skipped because the control panel will
+                    // not be rotating that quickly
```

Comment:
No LogKitten.wtf's should be committed to master—would recommend either making a lower level of LogKitten or omitting.